hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java

```diff
@@ -143,11 +142,12 @@ public class AWSGlueCatalogSyncClient extends HoodieSyncClient {
   private static final String ENABLE_MDT_LISTING = "hudi.metadata-listing-enabled";
   private final String databaseName;
 
-  private final Boolean skipTableArchive;
+  private final boolean skipTableArchive;
   private final String enableMetadataTable;
   private final int allPartitionsReadParallelism;
   private final int changedPartitionsReadParallelism;
   private final int changeParallelism;
+  private final Map<String, Table> initialTableByName = new HashMap<>();
```

Comment:
The cache works only for per-sync purpose right?

Reply:
Yes, it is just to reduce the number of times we lookup some current state from hive or glue to reduce the load.